File: srm.client/src/models/DeTaiData.js
import ModelBase from './ModelBase';

/**
 * @typedef {Object} DeTaiData
 * @property {string} [id] - Unique identifier
 * @property {string} [ngayTao] - Creation date
 * @property {string} [ngaySua] - Modification date
 * @property {string} [capDeTaiId] - Research level ID
 * @property {string} [tenCapDeTai] - Research level name
 * @property {string} [ten] - Title
 * @property {string} [maSo] - Code number
 * @property {string} [mucTieu] - Objective
 * @property {string} [noiDung] - Content
 * @property {number} [tongKinhPhi] - Total budget
 * @property {string} [ngayBatDau] - Start date
 * @property {string} [ngayKetThuc] - End date
 * @property {number} [kinhPhiHangNam] - Annual budget
 * @property {string} [hoSoNghiemThu] - Acceptance documents
 * @property {string} [hoSoSanPham] - Product documents
 * @property {string} [donViChuTriId] - Leading organization ID
 * @property {string} [tenDonViChuTri] - Leading organization name
 * @property {string} [chuNhiem] - Project leader
 * @property {string[]} [canBoThamGias] - Participating staff
 * @property {string} [phanChiaSuDongGop] - Contribution distribution
 */

/**
 * Research project data model
 */
const DeTaiData = {
  ...ModelBase,
  capDeTaiId: null,
  tenCapDeTai: null,
  ten: null,
  maSo: null,
  mucTieu: null,
  noiDung: null,
  tongKinhPhi: null,
  ngayBatDau: null,
  ngayKetThuc: null,
  kinhPhiHangNam: null,
  hoSoNghiemThu: null,
  hoSoSanPham: null,
  donViChuTriId: null,
  tenDonViChuTri: null,
  chuNhiem: null,
  canBoThamGias: [],
  phanChiaSuDongGop: null
};

export default DeTaiData;

File: SRM.Domain/Data/DeTaiData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SRM.Domain.Data
{
    public class DeTaiData : ModelBase
    {
        public Guid? CapDeTaiId { get; set; }
        public string? TenCapDeTai { get; set; }
        public string? Ten { get; set; }
        public string? MaSo { get; set; }
        public string? MucTieu { get; set; }
        public string? NoiDung { get; set; }
        public decimal? TongKinhPhi { get; set; }
        public DateTime? NgayBatDau { get; set; }
        public DateTime? NgayKetThuc { get; set; }
        public decimal? KinhPhiHangNam { get; set; }
        public string? HoSoNghiemThu { get; set; }
        public string? HoSoSanPham { get; set; }
        public Guid? DonViChuTriId { get; set; }
        public string? TenDonViChuTri { get; set; }
        public string? ChuNhiem { get; set; }
        public List<string>? CanBoThamGias { get; set; }
        public string? PhanChiaSuDongGop { get; set; }
    }
}
